### PR TITLE
Support for Phabricator

### LINKED ISSUE #75: Add phabricator third party provider
Phabricator is a useful PM tool that we have been using that also ships with an oauth 2.0 api. 

http://www.phabricator.com

### PR DESCRIPTION
**Pull request by hofan41**
Added support for third party installations of phabricator, the user must supply a URI in the config options in order for it to work.

Closes #75

### CODE REVIEW

File: lib/providers/phabricator.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    var settings = results.value;
+
+    return {
+        protocol: 'oauth2',
+        auth: settings.uri + '/oauthserver/auth/',
+        token: settings.uri + '/oauthserver/token/',
+        scope: ['whoami'],
+        scopeSeparator: ',',
+        profile: function (credentials, params, get, callback) {
+            var query = {
```
Comment (geek) [inline lib/providers/phabricator.js]: we add a newline after function declarations.  please update the PR with the newlines. Here is more detail on the styleguide: hapijs.com/styleguide

File: test/providers.js [hunk 2]
```diff
@@ -1689,4 +1689,104 @@ describe('#linkedin', function () {
...
+            var mock = new Mock.V2();
+            mock.start(function (provider) {
+
+                var server = new Hapi.Server();
+                server.connection({ host: 'localhost', port: 80 });
+                server.register(Bell, function (err) {
+
+                    expect(err).to.not.exist();
+
+                    expect( function() {
```
Comment (geek) [inline test/providers.js]: `expect(function () {`

Comment (geek) [inline test/providers.js]: you can rewrite as

`expect(Bell.providers.phabricator).to.throw(Error);`   <-- I think

Comment (hofan41) [inline test/providers.js]: I dig that. Updating now!

**Pull request merged**